File: server/src/main/java/org/jetbrains/bsp/bazel/server/bsp/utils/InternalAspectsResolver.java
package org.jetbrains.bsp.bazel.server.bsp.utils;

import io.vavr.Lazy;
import org.jetbrains.bsp.bazel.commons.Constants;
import org.jetbrains.bsp.bazel.server.bsp.info.BspInfo;

public class InternalAspectsResolver {

  private final BspInfo bspInfo;
  private final Lazy<String> prefix = Lazy.of(this::getPrefix);

  public InternalAspectsResolver(BspInfo bspInfo) {
    this.bspInfo = bspInfo;
  }

  public String resolveLabel(String aspect) {
    return prefix.get() + aspect;
  }

  public String getBazelBspRoot() {
    return this.bspInfo.bazelBspDir().toString();
  }

  private String getPrefix() {
    return "@" + Constants.ASPECT_REPOSITORY + "//:aspects.bzl%";
  }
}

File: server/src/main/kotlin/org/jetbrains/bsp/bazel/server/bsp/utils/InternalAspectsResolver.kt
package org.jetbrains.bsp.bazel.server.bsp.utils

import org.jetbrains.bsp.bazel.bazelrunner.BazelRelease
import org.jetbrains.bsp.bazel.commons.Constants
import org.jetbrains.bsp.bazel.server.bsp.info.BspInfo

class InternalAspectsResolver(val bspInfo: BspInfo, val bazelRelease: BazelRelease) {
    private val prefix: Lazy<String> = lazy { getPrefix() }
    fun resolveLabel(aspect: String): String = prefix.value + aspect

    val bazelBspRoot: String
        get() = bspInfo.bazelBspDir().toString()

    private fun getPrefix(): String = when (bazelRelease.major) {
        in 0..5 -> "@" + Constants.ASPECT_REPOSITORY + "//aspects:core.bzl%"
        else -> "@@" + Constants.ASPECT_REPOSITORY + "//aspects:core.bzl%"
    }
}
